libcloud/test/storage/test_runabove.py

```diff
@@ -0,0 +1,243 @@
+            pass
+        else:
+            self.fail('Exception was not thrown')
+
+    def test_get_container(self):
+        container = self.driver.get_container('foo', ex_location='BHS-1')
+        self.assertEqual('foo', container.name)
+
+    def test_get_container_not_found(self):
+        try:
+            self.driver.get_container('not_found', ex_location='BHS-1')
+        except ContainerDoesNotExistError:
+            pass
+        else:
+            self.fail('Exception was not thrown')
+
+    def test_get_object(self):
+        obj = self.driver.get_object('foo', 'foo_file', ex_location='BHS-1')
+        self.assertEqual('foo_file', obj.name)
+
+    def test_get_object_not_found(self):
+        try:
+            self.driver.get_object('foo', 'not_found', ex_location='BHS-1')
+        except ObjectDoesNotExistError:
+            pass
+        else:
+            self.fail('Exception was not thrown')
+
+    def test_get_object_but_container_not_found(self):
+        try:
+            self.driver.get_object('not_found', 'foo_file', ex_location='BHS-1')
+        except ContainerDoesNotExistError:
+            pass
+        else:
+            self.fail('Exception was not thrown')
+
+    def test_create_container(self):
+        container = self.driver.create_container('new_foo', ex_location='BHS-1')
+        self.assertEqual('new_foo', container.name)
+
+    def test_delete_container(self):
+        container = RunAboveContainer(name='foo', extra={'region': 'BHS-1'}, driver=self.driver)
+        succeed = self.driver.delete_container(container)
+        self.assertTrue(succeed)
+
+    def test_delete_not_found_container(self):
+        container = RunAboveContainer(name='not_found', extra={'region': 'BHS-1'}, driver=self.driver)
+        try:
+            self.driver.delete_container(container)
+        except ContainerDoesNotExistError:
+            pass
+        else:
+            self.fail('Exception was not thrown')
+
+    def test_delete_not_empty_container(self):
+        container = RunAboveContainer(name='not_empty', extra={'region': 'BHS-1'}, driver=self.driver)
+        try:
+            self.driver.delete_container(container)
+        except ContainerIsNotEmptyError:
+            pass
+        else:
+            self.fail('Exception was not thrown')
+
+    def test_upload_object(self):
+        test_file_path = '/tmp/foo'
+        with open(test_file_path, 'w') as fd:
+            fd.write('FOO')
+        container = RunAboveContainer(name='foo', extra={'region': 'BHS-1'}, driver=self.driver)
+        self.driver.upload_object(test_file_path, container, 'foo_file')
+
+    def test_upload_object_via_stream(self):
+        test_file = StringIO('bar')
+        container = RunAboveContainer(name='foo', extra={'region': 'BHS-1'}, driver=self.driver)
+        obj = self.driver.upload_object_via_stream(test_file, container, 'foo_file')
+        self.assertEqual('foo_file', obj.name)
+
+    # def test_download_object(self):
```

Comment:
The test should work be raise me `RuntimeError: maximum recursion depth exceeded while calling a Python object`:

```
ERROR: test_download_object (__main__.RunAboveStorageTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "libcloud/test/storage/test_runabove.py", line 217, in test_download_object
    data = self.driver.download_object(obj, '/tmp/foo2')
  File "/usr/local/lib/python2.7/dist-packages/apache_libcloud-0.18.0-py2.7.egg/libcloud/storage/drivers/runabove.py", line 266, in download_object
    callback_kwargs={'response': response.response, 'obj': obj,
  File "/usr/local/lib/python2.7/dist-packages/apache_libcloud-0.18.0-py2.7.egg/libcloud/common/base.py", line 266, in response
    if not self.success():
  File "/usr/local/lib/python2.7/dist-packages/apache_libcloud-0.18.0-py2.7.egg/libcloud/common/base.py", line 179, in success
    return self.status in [httplib.OK, httplib.CREATED]
  File "/usr/local/lib/python2.7/dist-packages/apache_libcloud-0.18.0-py2.7.egg/libcloud/common/base.py", line 273, in status
    self._status = self.response.status
  File "/usr/local/lib/python2.7/dist-packages/apache_libcloud-0.18.0-py2.7.egg/libcloud/common/base.py", line 266, in response
    if not self.success():
  File "/usr/local/lib/python2.7/dist-packages/apache_libcloud-0.18.0-py2.7.egg/libcloud/common/base.py", line 179, in success
    return self.status in [httplib.OK, httplib.CREATED]
  File "/usr/local/lib/python2.7/dist-packages/apache_libcloud-0.18.0-py2.7.egg/libcloud/common/base.py", line 273, in status
    self._status = self.response.status
  File "/usr/local/lib/python2.7/dist-packages/apache_libcloud-0.18.0-py2.7.egg/libcloud/common/base.py", line 266, in response
    if not self.success():
...
```

I tested manually, it works but I'm not able to make unit test with ``_get_object`` mechanism (some help would be appreciated)